build-support/scripts/protobuf.sh

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env bash
+
+readonly SCRIPT_NAME="$(basename ${BASH_SOURCE[0]})"
+readonly SCRIPT_DIR="$(dirname "${BASH_SOURCE[0]}")"
+readonly SOURCE_DIR="$(dirname "$(dirname "${SCRIPT_DIR}")")"
+readonly FN_DIR="$(dirname "${SCRIPT_DIR}")/functions"
+
+source "${SCRIPT_DIR}/functions.sh"
+
+unset CDPATH
+
+set -euo pipefail
+
+usage() {
```

Comment:
I have to adjust this usage to be more correct. I cribbed this from `proto-gen.sh` originally.